kolibri_instant_schools_plugin/assets/src/views/sign-up-page/index.vue

```diff
@@ -75,22 +74,36 @@
         v-model="confirmedPassword"
       />
 
-      <ui-select
-        :name="$tr('selectFacility')"
-        :placeholder="$tr('selectFacility')"
-        :label="$tr('facility')"
-        :value="selectedFacility"
-        :options="facilityList"
-        :invalid="facilityIsInvalid"
-        :error="facilityIsInvalidText"
-        @blur="facilityBlurred = true"
-        @input="updateSelection"
+      <!-- ToS modal only shows up if the box empty prior to clicking -->
+      <!-- Using UI Checkbox for blur events and v-model -->
+      <k-checkbox
+        id="terms-agreement-checkbox"
+        :class="['terms-agreement-checkbox', termsNotAgreed ? 'invalid' : '']"
+        :checked="termsAgreed"
+        @change="showTerms = termsAgreed = $event"
+        @blur="termsAgreementCheckboxBlurred = true"
+        :label="$tr('termsAgreementLabel')"
       />
+      <label
+        v-if="termsNotAgreed"
+        for="terms-agreement-checkbox"
+        class="terms-error-box"
+      >
+        {{ termsNotAgreedText }}
+      </label>
 
       <k-button :disabled="busy" :primary="true" :text="$tr('finish')" type="submit" />
 
     </form>
 
+    <core-modal
+      v-if="showTerms"
```

Comment:
The UX of this checkbox/modal selection is weird. How about this instead? Upon clicking the checkbox/label, the modal is opened. The modal then contains 2 buttons, [decline] and [accept]. Clicking decline closes the modal and the checkbox remains unselected. Clicking accept, closes the modal and the checkbox is selected. It's still unconventional but I think a better experience. Thoughts @jtamiace

Reply:
A pattern I see often is something like:

* [ ] I agree to the terms. [View terms]()

where the the 'view' link and the checkbox label are independently actions.

Reply:
Is there a requirement that users HAVE to view the terms of service? If not, then yes this is much better. We would just have to add a slot to the k-checkbox.

Reply:
> Is there a requirement that users HAVE to view the terms of service?

I don't believe so. Even if they "view" it, did they scroll to the end? Did they read it? Did they understand it? (cc @jeepurs feel free to weigh)

> We would just have to add a slot to the k-checkbox

Might be easier to just do

    [checkbox component] [link]

as two inline blocks rather than changing the API of the checkbox

Reply:
> as two inline blocks rather than changing the API of the checkbox

Reply:
The API change (adding blur) is more for validation than including links, though?

Reply:
+1 on @indirectlylit's suggestion. checking box and viewing terms should be two separate actions. An alternative is to make the entire "I agree to the terms and service" thing clickable but don't know what's best dev-wise

Reply:
I don't think we need a requirement that users HAVE to view the TOS @christianmemije

Reply:
Cool! Then let's go with @indirectlylit's suggestion @DXCanas. :tada:

Reply:
Had to go a bit of a different route due to the checkbox's markup structure not allowing for reflow that way. See https://github.com/fle-internal/kolibri-instant-schools-plugin/pull/65#issuecomment-341251636